test/integration/test_connection_management.py

```diff
@@ -105,3 +105,36 @@ def test_http_h2_connection_management_single_request_per_conn_1(http_test_serve
 @pytest.mark.skipif(isSanitizerRun(), reason="Unstable in sanitizer runs")
 def test_http_h2_connection_management_single_request_per_conn_1(http_test_server_fixture):
   connection_management_test_request_per_connection(http_test_server_fixture, 5, True)
+
+
+@pytest.mark.skipif(isSanitizerRun(), reason="Unstable in sanitizer runs")
+def test_h1_pool_strategy(http_test_server_fixture):
+  """
+  Test that with the "HOT" strategy only the first created connection gets to send requests.
+  Then, with the "FAIR" strategy, we expect the other connection to be used as well.
+  """
+
+  def countLogLinesWithSubstring(logs, substring):
+    return len([line for line in logs.split(os.linesep) if substring in line])
+
+  _, logs = http_test_server_fixture.runNighthawkClient([
+      "--rps 20", "-v", "trace", "--connections", "2", "--prefetch-connections",
+      "--h1-connection-reuse-strategy", "HOT", "--termination-predicate", "benchmark.http_2xx:10",
+      http_test_server_fixture.getTestServerRootUri()
+  ])
+
+  requests = 60
+  connections = 3
+  assertNotIn("[C1] message complete", logs)
+  assertGreaterEqual(countLogLinesWithSubstring(logs, "[C0] message complete"), 10)
+
+  _, logs = http_test_server_fixture.runNighthawkClient([
+      "--rps", "20", "-v trace", "--connections",
+      str(connections), "--prefetch-connections", "--h1-connection-reuse-strategy", "FAIR",
+      "--termination-predicate",
+      "benchmark.http_2xx:%d" % requests,
+      http_test_server_fixture.getTestServerRootUri()
+  ])
+  for i in range(1, connections):
+    assertGreaterEqual(
```

Comment:
Is an upper bound also possible to specify?

Reply:
Yes, it was possible to tighten this up, done in bf86ebee79eccd3195bc22c701adb8977516d23f

Reply:
Looks good, thanks.